src/seedu/addressbook/data/person/Address.java

```diff
@@ -52,8 +51,5 @@ public boolean equals(Object other) {
     public int hashCode() {
         return value.hashCode();
```

Comment:
The main motivation of inheritance is to reduce duplicity of code. In these files, methods like hashCode(), Validation, etc. are redundant across the files. Please try to implement these generic functions in the parent class.